tests/models_test.py

```diff
@@ -96,25 +96,33 @@ def test_no_top_variable_shape_4(app, last_feature_dim):
 
 
 @parametrize
-def test_model_summary(app, last_feature_dim, snapshot):
+def test_model_summary(app, last_feature_dim):
     input_tensor = keras.layers.Input(shape=(224, 224, 3))
     model = app(weights=None, input_tensor=input_tensor)
-    json = str(model.to_json())
+
+    class PrintToVariable:
+        output = ""
+
+        def __call__(self, x):
+            self.output += f"{x}\n"
+
+    capture = PrintToVariable()
+    lq.models.summary(model, print_fn=capture)
```

Comment:
```suggestion
    import io
    from contextlib import redirect_stdout

    capture = io.StringIO()
    with redirect_stdout(f):
        lq.models.summary(model)
```

You can then use `capture.getvalue()`.

I want to refactor `lq.models.summary()` to remove the `print_fn` function anyway since I think it's a bit of an antipattern when we can use a built-in wrapper like this to achieve the same result.

Reply:
Never mind, using the `print_fn` is actually better because the `redirect_stdout` would break if we have other random (TF) prints happening at some points.